http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamServiceInterp.scala

```diff
@@ -0,0 +1,138 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao.sam
+
+import cats.effect.Async
+import cats.mtl.Ask
+import cats.syntax.all._
+import com.google.api.services.storage.StorageScopes
+import org.broadinstitute.dsde.workbench.azure.AzureCloudContext
+import org.broadinstitute.dsde.workbench.client.sam.ApiException
+import org.broadinstitute.dsde.workbench.client.sam.model.GetOrCreateManagedIdentityRequest
+import org.broadinstitute.dsde.workbench.leonardo.auth.CloudAuthTokenProvider
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoInternalServerError
+import org.broadinstitute.dsde.workbench.leonardo.{AppContext, SamResourceType, WorkspaceId}
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import org.broadinstitute.dsde.workbench.model.{UserInfo, WorkbenchEmail}
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import org.http4s.AuthScheme
+import org.http4s.Credentials.Token
+import org.typelevel.log4cats.StructuredLogger
+
+import java.util.UUID
+import scala.jdk.CollectionConverters._
+
+class SamServiceInterp[F[_]](apiClientProvider: SamApiClientProvider[F],
+                             cloudAuthTokenProvider: CloudAuthTokenProvider[F]
+)(implicit
+  F: Async[F],
+  metrics: OpenTelemetryMetrics[F],
+  logger: StructuredLogger[F]
+) extends SamService[F] {
+
+  private val saScopes = Seq(
+    "https://www.googleapis.com/auth/userinfo.email",
+    "https://www.googleapis.com/auth/userinfo.profile",
+    StorageScopes.DEVSTORAGE_READ_ONLY
+  )
+
+  // TODO: pattern for retries
```

Comment:
Do you want to tackle this as part of this PR? It does not have to, if you want to create a follow up ticket in the same epic as IA-4327, or a tech debt related to IA-4295 that works too

Reply:
I'll make a tech debt ticket related to [IA-4295](https://broadworkbench.atlassian.net/browse/IA-4295)

Reply:
https://broadworkbench.atlassian.net/browse/IA-5053, will reference it in a comment as well